invokeai/backend/stable_diffusion/extensions/lora_patcher.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Dict, Iterator, List, Optional, Tuple
+
+import torch
+from diffusers import UNet2DConditionModel
+
+from invokeai.backend.stable_diffusion.extensions.base import ExtensionBase
+from invokeai.backend.util.devices import TorchDevice
+
+if TYPE_CHECKING:
+    from invokeai.app.invocations.model import LoRAField
+    from invokeai.app.services.shared.invocation_context import InvocationContext
+    from invokeai.backend.lora import LoRAModelRaw
+
+
+class LoRAPatcherExt(ExtensionBase):
+    def __init__(
+        self,
+        node_context: InvocationContext,
+        loras: List[LoRAField],
+        prefix: str,
+    ):
+        super().__init__()
+        self._loras = loras
+        self._prefix = prefix
+        self._node_context = node_context
+
+    @contextmanager
+    def patch_unet(self, unet: UNet2DConditionModel, cached_weights: Optional[Dict[str, torch.Tensor]] = None):
+        def _lora_loader() -> Iterator[Tuple[LoRAModelRaw, float]]:
+            for lora in self._loras:
+                lora_info = self._node_context.models.load(lora.lora)
+                lora_model = lora_info.model
+                yield (lora_model, lora.weight)
+                del lora_info
+            return
+
+        yield self._patch_model(
+            model=unet,
+            prefix=self._prefix,
+            loras=_lora_loader(),
+            cached_weights=cached_weights,
+        )
+
+    @classmethod
+    @contextmanager
+    def static_patch_model(
```

Comment:
Unused?

Reply:
See comment about `_patch_model`.
Shortly - will be used by compel nodes when we test that output fully same(I locally tested, just want to someone else test too)